code/weapon/weapons.cpp

```diff
@@ -7497,7 +7524,55 @@ void weapon_hit( object * weapon_obj, object * other_obj, vec3d * hitpos, int qu
 		weapon_hit_do_sound(other_obj, wip, hitpos, armed_weapon, quadrant);
 	}
 
-	if (wip->impact_weapon_expl_effect.isValid() && armed_weapon) {
+
+ 	bool valid_conditional_impact = false;
+	int relevant_armor_idx = -1;
+	float relevant_fraction = 1.0f;
+	float hit_angle = 0.0f;
+	vec3d reverse_incoming = (vmd_zero_vector - weapon_obj->orient.vec.fvec);
+
+	if (hitnormal) {
+		hit_angle = vm_vec_delta_ang(hitnormal, &reverse_incoming, NULL);
+	}
+
+	if (other_obj != nullptr && other_obj->type == OBJ_SHIP) {
```

Comment:
This it the short circuiting you're talking about?  It looks fine to me.  It's just a general safety thing to check that the object is not null before dereferencing, tbh.  But if it's already guaranteed to be a non-null pointer, then you don't need to check.